src/proto/rng.rs

```diff
@@ -0,0 +1,60 @@
+//! `Rng` protocol.
+
+use crate::{data_types::Guid, proto::Protocol, unsafe_guid, Result, Status};
+use core::ptr;
+
+/// Contain a Rng algorithm Guid
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub struct RngAlgorithm(pub Guid);
```

Comment:
Let's model this as a newtype enum. See [`GptPartitionType`](https://github.com/rust-osdev/uefi-rs/blob/main/src/proto/media/partition.rs#L51) for an example of what would look like. We could then add constants for the guids in 37.5.1.

Reply:
I exactly though about this example for a better implementation.

Reply:
Done